switch/switch_main.ml

```diff
@@ -176,9 +176,17 @@ let make_server config =
       >>= fun qs ->
       on_disk_queues := qs;
       info "Reading the redo-log from %s" redo_log_path;
-      let open Lwt_result in
-      Block.connect ("buffered:" ^ redo_log_path)
+      Block.connect ~buffered:true redo_log_path
+      >>= fun block -> begin
+        match block with
+        | `Ok block -> Lwt.return block
+        | `Error e ->
```

Comment:
A couple of suggestions:

* Avoid a nested match by matching `Ok block`, `Error Unknown` and `Error _` in one match.
* You can provide a line number as in `Lwt.fail (Failure (Printf.sprintf "Other error in %s" __LOC__))`